cmake/platform.cmake

```diff
@@ -290,6 +290,15 @@ if(MSVC)
         append(CMAKE_CCXX_FLAGS "-Wno-unknown-warning-option")
     endif()
 elseif(UNIX OR MINGW)
+
+    # Enable debug checks for common C++ standard libraries
+    if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        # Enable debug checks for libstdc++
+        append(CMAKE_CCXX_FLAGS "-D_GLIBCXX_DEBUG")
```

Comment:
According to libstdc++ documentation this flag breaks ABI for some classess:
> Note that this flag changes the sizes and behavior of standard class templates such as std::vector, and therefore you can only link code compiled with debug mode and code compiled without debug mode if no instantiation of a container is passed between the two translation units.

Will this create issues for debug builds of applications using oneDNN?

Reply:
Yeah, this does seem like it could cause problems. Would it be reasonable to guard this behind `ONEDNN_DEV_MODE` which is not generally used by applications?

Reply:
Sounds reasonable.